pkg/provider/configure_clients.go

```diff
@@ -142,6 +152,69 @@ func createGrafanaAPIClient(client *common.Client, providerConfig ProviderConfig
 	return nil
 }
 
+func createGrafanaAppPlatformClient(client *common.Client, cfg ProviderConfig) error {
+	rcfg := rest.Config{
+		UserAgent: cfg.UserAgent.ValueString(),
+		Host:      cfg.URL.ValueString(),
+		APIPath:   "/apis",
+	}
+
+	tlsClientConfig, err := parseTLSconfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	// Kubernetes really is wonderful, huh.
+	// tl;dr it has it's own TLSClientConfig,
+	// and it's not compatible with the one from the "crypto/tls" package.
+	rcfg.TLSClientConfig = rest.TLSClientConfig{
+		Insecure: tlsClientConfig.InsecureSkipVerify,
+	}
+
+	if len(tlsClientConfig.CertData) > 0 {
+		rcfg.CertData = tlsClientConfig.CertData
+	}
+
+	if len(tlsClientConfig.KeyData) > 0 {
+		rcfg.KeyData = tlsClientConfig.KeyData
+	}
+
+	if len(tlsClientConfig.CAData) > 0 {
+		rcfg.CAData = tlsClientConfig.CAData
+	}
+
+	userInfo, orgID, apiKey, err := parseAuth(cfg)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case apiKey != "":
+		if orgID > 1 {
+			return fmt.Errorf("org_id is only supported with basic auth. API keys are already org-scoped")
+		}
+
+		rcfg.BearerToken = apiKey
+	case userInfo != nil:
+		rcfg.Username = userInfo.Username()
+		if p, ok := userInfo.Password(); ok {
+			rcfg.Password = p
+		}
+	}
+
+	client.GrafanaOrgID = cfg.OrgID.ValueInt64()
+	client.GrafanaStackID = cfg.StackID.ValueInt64()
+	client.GrafanaAppPlatformAPIClientID = cfg.UserAgent.ValueString()
+	client.GrafanaAppPlatformAPI = k8s.NewClientRegistry(rcfg, k8s.ClientConfig{
+		// TODO: check with @IfSentient if we should be using this serializer provider.
+		// NegotiatedSerializerProvider: func(kind resource.Kind) runtime.NegotiatedSerializer {
+		// 	return &k8s.KindNegotiatedSerializer{Kind: kind}
+		// },
```

Comment:
Did you check with @IfSentient ? :)

Reply:
Ah, good catch – I don't think I did.

Reply:
I think it should be ok to use the `KindNegotitatedSerializer` here, all local, low-scale testing I've done has done fine.

Reply:
Updated.